paella-opencast/plugins/es.upv.paella.opencast.loader/01_prerequisites.js

```diff
@@ -155,7 +155,18 @@ paella.opencast = new (Class ({
             if (authResult && authResult.dceReturnStatus) {
                 var returnStatus = authResult.dceReturnStatus;
                 if (("401" == returnStatus || "403" == returnStatus) && authResult.dceLocation) {
+
+                  var redirectDelay = window.parent !== window
+                    ? $("iframe", window.parent.document).index(self.frameElement) * 100
```

Comment:
@lbjay If you  set up a sample Canvas page that mimics the issue into my Canvas sandbox, I'll load this onto kdolan-dev and QA will be able to verify all scenarios.